### PR TITLE
Api 22214/Handling APIVersion updates

### PR DESCRIPTION
**Pull request by ediiotero**
Story: [API-22214](https://vajira.max.gov/browse/API-22214)

Added a new apiVersionPUTSchema and logic to prevent changing majorVersion or fhirVersion when updating an API version

### CODE REVIEW

File: src/api/version/apiVersion.schema.ts [hunk 1]
```diff
@@ -72,3 +72,9 @@ export const apiVersionLookupSchema = Joi.object()
     version: Joi.string().required(),
   })
   .options({ abortEarly: false });
+
+export const apiVersionPUTSchema = apiVersionMutateSchema
```
Comment (mheusgit) [inline src/api/version/apiVersion.schema.ts:76]: Haven't seen this done before.  This approach should help reduce number of lines in other schemas in the near future.

Comment (mheusgit) [reply src/api/version/apiVersion.schema.ts:76]: Did a quick check.  The other schemas wont benefit from this as much as I'd hoped.

**Comment by mheusgit** [comment]
Pulled this logic down for testing.  I noticed it failed to detect version updates if I moved from majorVersion "1" to "0"

File: src/api/version/apiVersion.service.ts [hunk 2]
```diff
@@ -117,7 +117,32 @@ class APIVersionService {
   public async updateAPIVersion(
     model: APIVersionModel,
     updates: Optional<APIVersion>,
+    version: string,
   ): Promise<APIVersionModel> {
+    if (updates.majorVersion || updates.fhirVersion) {
+      const majorVersion = updates.majorVersion
```
Comment (mheusgit) [inline src/api/version/apiVersion.service.ts:123]: I suspect "updates.majorVersion ?" if evaluating to false when majorVersion is "0" but the intent was to check if it was undefined

File: src/api/version/apiVersion.service.ts [hunk 3]
```diff
@@ -117,7 +117,32 @@ class APIVersionService {
   public async updateAPIVersion(
     model: APIVersionModel,
     updates: Optional<APIVersion>,
+    version: string,
   ): Promise<APIVersionModel> {
+    if (updates.majorVersion?.toString() || updates.fhirVersion) {
```
Comment (dsteeve) [inline src/api/version/apiVersion.service.ts:122]: would it make sense to add a test to try to change the api version from 1 to 0 to make sure this code block is exercised appropriately?

**Review by dsteeve — COMMENTED** [review]
might want to add additional testing to cover the 0 case Marc mentions.

**Review by mheusgit — APPROVED** [review]
Looks good.  Requested changes appear to be in place

**Review by stevelong00 — COMMENTED** [review]
Looks good!

**Pull request merged**